osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs

```diff
@@ -269,21 +269,50 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
 
-            Playlist.BindCollectionChanged(onPlaylistChanged, true);
+            matchCurrentItem.BindTo(client.CurrentMatchPlayingItem);
+            matchCurrentItem.BindValueChanged(onCurrentItemChanged, true);
+
             BeatmapAvailability.BindValueChanged(updateBeatmapAvailability, true);
             UserMods.BindValueChanged(onUserModsChanged);
 
             client.LoadRequested += onLoadRequested;
             client.RoomUpdated += onRoomUpdated;
 
-            isConnected = client.IsConnected.GetBoundCopy();
+            isConnected.BindTo(client.IsConnected);
             isConnected.BindValueChanged(connected =>
             {
                 if (!connected.NewValue)
                     Schedule(this.Exit);
             }, true);
         }
 
+        private void onCurrentItemChanged(ValueChangedEvent<PlaylistItem> item)
+        {
+            if (client?.LocalUser == null)
+                return;
+
+            // If we're about to enter gameplay, schedule the item to be set at a later time.
+            if (client.LocalUser.State > MultiplayerUserState.Ready)
```

Comment:
Is there a reason to only schedule in this scenario rather than always?

Reply:
This is all technically not required anymore. It existed because the server previously set `expired = 1` and a new playlist item on match start, rather than match end. I wanted to make sure that the current item didn't change as a result of that, while the client is starting the match.

Not required anymore because the server only does this when the match ends now.